### ISSUE TITLE
Accessibility: Polymer modal dialogs should make the rest of the UI inert and aria-hidden

### ISSUE THREAD

**Issue by TimvdLippe** [PolymerElements/paper-dialog-behavior#125]
_From @minorninth on March 20, 2018 15:56_

When a modal dialog or other pop-up appears that visually covers the rest of the web page, two things must happen in order for it to be fully accessible:

* The rest of the content should be hidden using aria-hidden=true on each subtree other than the one containing the dialog.
* Any focusable controls should be disabled or otherwise made inert / non-focusable.

I believe all other web frameworks do this already (e.g. Google's Closure/Wiz, AngularUI, React).

Finally, when the modal is closed, ideally focus should be restored to wherever it was last.

To repro, open this demo and click "Modal Dialog":

  https://www.webcomponents.org/element/PolymerElements/paper-dialog/demo/demo/index.html

Then with a screen reader enabled, use the screen reader keys to navigate the rest of the page. It's very easy to explore content outside of the dialog.

For example, with VoiceOver use Ctrl+Option+Left/Right to navigate and Ctrl+Option+Shift+Up to "escape" the dialog. (This shouldn't be possible.)

_Copied from original issue: Polymer/polymer#5156_

**Comment by valdrinkoshi** [comment]
We provide a solution to this issue with https://github.com/PolymerLabs/blocking-elements, a polyfill for the Blocking Elements stack API proposal that allows to inert all but one node, the top blocking element. Examples on how to use it here https://codelabs.developers.google.com/codelabs/building-custom-overlays/#3
I'm currently working on publishing it on npm.
